src/infrastructure/amqp/ConsumerErrorResolver.ts

```diff
@@ -0,0 +1,35 @@
+import { InternalError } from '@lokalise/node-core'
+import { ZodError } from 'zod'
+
+import type { ErrorResolver } from '../errors/ErrorResolver'
+import { isStandardizedError } from '../typeUtils'
+
+import { AmqpMessageInvalidFormat, AmqpValidationError } from './amqpErrors'
+
+export class ConsumerErrorResolver implements ErrorResolver {
+  public processError(error: unknown): InternalError {
+    if (error instanceof Error && error?.name === 'SyntaxError') {
+      return new AmqpMessageInvalidFormat({
+        message: error.message,
+      })
+    }
+    if (error instanceof ZodError) {
```

Comment:
:large_orange_diamond: Can we expect from `Zod` some kind of versioning mechanism for incoming events?

Or the way it should be: "always keep events backwards compatible".

Reply:
Thank you Igor, fair enough :blush: